core/ledger/kvledger/txmgmt/txmgr/query_executor.go

```diff
@@ -214,9 +212,10 @@ func (q *queryExecutor) GetPrivateData(ns, coll, key string) ([]byte, error) {
 		return nil, err
 	}
 	if !version.AreSame(hashVersion, ver) {
-		return nil, &ErrPvtdataNotAvailable{Msg: fmt.Sprintf(
+		return nil, errors.Errorf(
```

Comment:
Does it return a stack trace to the chaincode invoker or just the error message? Same question for other log messages too. I think just the error message is sufficient for the end-user.

Reply:
We shouldn't be worrying at this deep level about this. There are many layers above this as well and keeping track at each level is not a good way. The last layer should propagate just the string and wherever in between we log, the stack is printed. I checked in the chaincode package [here](https://github.com/hyperledger/fabric/blob/24868e956881c604d7dc1b561ce3f18ce2fc62b5/core/chaincode/handler.go#L250) and at that level it returns the string only `err.Error()`

Reply:
I see. Make sense.